django/http/response.py

```diff
@@ -333,6 +334,17 @@ def __repr__(self):
             'content_type': self._content_type_for_repr,
         }
 
+    def __getstate__(self):
+        # neglect non-pickleable attributes in response
+        state = {}
+        for k, v in self.__dict__.items():
+            try:
+                pickle.dumps(v)
+                state[k] = v
+            except Exception:
+                pass
+        return state
```

Comment:
I can certainly appreciate the simplicity of doing it this way, and perhaps I even like it for that ... but I think it's also maybe too ambiguous (as-is). The ticket only _calls_ for consuming and restoring 1) `.content` 2) `.headers` 3) and `.status_code` ... (though I suspect `.charset` would also be a sensible candidate) 

As [currently] written, the benefit and detriment is that anything _could_ end up in the pickled output. For some that would probably be a boon, for others it might be unexpected and undesired (eg: `cookies`, private attributes etc) but there's a degree of precedent for doing it the way you have ... ish. If you look at `TemplateResponse.__getstate__` it already implements a similar thing, but with additional blacklisting certain variables. We should probably do something similar here, so that at least it's known what _won't_ get serialised. Bonus points if we could refactor both into one implementation?

Reply:
Thanks for reviewing. I have thought about pickling only  content, headers and status_code as the ticket suggests, but I think some may find it confusing when they put something in HTTPResponse and lost it after pickling it. 

Maintaining a blacklist is a good idea, it avoid the ambiguity in my implementation and won't lose any information  we don't want to. I will fix it shortly.